File: Basic/Math/reverse_num.py
def reverse(x):
    rev = 0
    while x != 0:
        digit = x % 10
        if rev > 214748364 or (rev == 214748364 and digit > 7):
            return 0
        if rev < -214748364 or (rev == -214748364 and digit < -8):
            return 0
        rev = rev * 10 + digit
        x //= 10
    return rev

File: Basic/Math/ReverseNum.cpp
class ReverseNum {
public:
    static int reverse(int x) {
        long long reversed = 0;
        while (x != 0) {
            reversed = reversed * 10 + x % 10;
            if (reversed > 2147483647LL || reversed < -2147483648LL) return 0;
            x /= 10;
        }
        return static_cast<int>(reversed);
    }
};